### PR TITLE
feat(ec2): add ICMPv6 protocol 

### PR DESCRIPTION
**Pull request by philipmw**
This allows creation of a security group rule that permits
ICMPv6.
----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
Thank you for your contribution. We cannot review PRs that do not include tests.

**Comment by philipmw** [comment]
@TheRealAmazonKendra , my change has no logic. That file has no corresponding unit test file, and the surrounding (existing) methods have no unit tests. (Unless I missed them.) So, my PR is following the existing convention. Would you please reconsider? Else please advise on a meaningful unit test for this change.

**Comment by philipmw** [comment]
@TheRealAmazonKendra - ping.

**Comment by TheRealAmazonKendra** [comment]
The lack of tests for this file is something that needs to be remedied. We should not have an entire file of functions that are untested. In your case, you should write a test that verifies that your function creates the template you'd expect it to create. Note that the creation of a new function also requires an update to the README. Right now it says we don't have implementations for ICMP. I will however, exempt integ tests.

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
The addition of new functions must always have tests. We shouldn't have approved these in the past without them. We need to ensure that the template generated as a result of this function is what you expect it to be. Note that the linter will also fail if no README update is made as well. [Port Ranges](https://github.com/aws/aws-cdk/tree/main/packages/%40aws-cdk/aws-ec2#port-ranges) should be updated to reflect this change.

**Comment by philipmw** [comment]
Thanks, @TheRealAmazonKendra . I updated the PR with a new unit test and an updated README.

File: packages/@aws-cdk/aws-ec2/test/port.test.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { Port, Protocol } from '../lib';
+
+describe('port', () => {
+  describe('Port class', () => {
+    describe('.allIcmpV6', () => {
+      it('returns expected instance of Port', () => {
+        expect(Port.allIcmpV6()).toEqual(new Port({
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/aws-ec2/test/port.test.ts:7]: Let's check this against the template generated, not against the data structure itself.

**Comment by TheRealAmazonKendra** [comment]
This PR has been in the CHANGES REQUESTED state for 3 weeks, and looks abandoned. To keep this PR from being closed, please continue work on it. If not, it will automatically be closed in a week.

**Review by TheRealAmazonKendra — APPROVED** [review]
My apologies for not being more clear on what I was asking for in the test. A unit test comparing the output to the produced template would have been sufficient but this is great too. It was probably just more work for you than I intended.

Still, thanks for your hard work on this and for contributing!

**Pull request merged**